SMLHelper/Patchers/LanguagePatcher.cs

```diff
@@ -191,7 +191,7 @@ internal static void AddCustomLanguageLine(string modAssemblyName, string lineId
 
         internal static string TrimTextDelimiters(string value)
         {
-            return value.Trim(TextDelimiterOpen, TextDelimiterClose);
+            return value.Substring(1, value.Length - 2);
```

Comment:
Starting a new thread:  

With this change that trims out the outer delimiters by index, all that would be left would be to account for `string.Format` tokens `{0}` and Unix-style line break characters (which would be picked up by Windows) `\n`

Regex replacement could be done to convert patterns like `{0} {1} {2}` into `[0] [1] [2]` or something equally simple.  
And the same could be done to replace the line break characters `\n` with something like `[NewLine]`.

This wouldn't be too hard to handle in each direction, and would require a minimal level of updates on behalf of the user to convert their existing files to use these new tokens instead.

Reply:
I personally would take a different approach, by not escaping the characters at all, but by changing the way the file is parsed. Instead of checking each file individually we can parse it using regex and checking for the delimiter characters `:{` and `}\n`

Reply:
That should be doable. Here's a regex to try.  
`(?<key>[\w]+):(?<value>{[\w\s{}\n]+}\n\r?) `  
We'll need to make sure that multi-line matching is enabled

Reply:
Better regex: `(?<key>[\w]+):\s*{(?<value>({\d+}|[\w\s\n])+)}(\n|\n\r)*`

Reply:
Even better regex: `(?<key>\w+?)\s*?:\s*?{(?<value>[\s\S]+?)}(\n|\n\r)?`

Reply:
`\s\S` is too lazy. This regex fails to handle the string.format tokens.
Sample that doesn't work for this one `key:{val{0}ue}`

Reply:
Until we can find a better way,  
`(?<key>[\w]+)\s*:\s*{(?<value>([~!@#$%^&*()\-_=+\[\];:"',<>\/?]|{\d+}|[\w\s\n])+)}(\n|\n\r)*`  
this regex includes explicit matching on every special character I can easily access from my keyboard.

Matches special characters `[~!@#$%^&*()\-_=+\[\];:"',<>\/?]`  
Matches string.format tokens `{\d+}`
Matches letters, numbers, and line breaks `[\w\s\n]`

Reply:
`(?<key>\w+?)\s*?:\s*?{(?<value>[\s\S]+?)}(\n|\n\r)$`

Reply:
That somehow works against my test text
```
key:{value}
key:{valu{0}e}
key:{valu
{0}e}
key:{val-u{0}e}
key:{val__[!]
-u{0}{1}e}
```
Validated on https://regex101.com/

Reply:
`$` matches the end of the text/line so we don't have to check for `\n` and `\r` characters

Reply:
The groups are also parsed correctly

![image](https://user-images.githubusercontent.com/32238504/57935116-c0357800-78c9-11e9-8c91-642e7456444d.png)

Reply:
Looks promising.  
If you want to try replacing the line-by-line parsing logic with one that applies this regex to the full text and then iterates over the matches, go for it.

Reply:
Still the only regex that passes all tests  
`(?<key>[\w]+)\s*:\s*{(?<value>([~!@#$%^&*()\-_=+\[\];:"',<>\/?]|{\d+}|[\w\s\n])+)}(\n|\r\n)*`